### ISSUE TITLE
Incompatibility with new major release of hashicorp/google-beta

### ISSUE THREAD

**Issue by zdenekvana** [basisai/terraform-google-vault#10]
Hi, there are some issues with latest release of hashicorp/google-beta provider. It would be nice to add there at least version lock like e.g. ">= 3.70, < 4.0.0" till the fix.

```
│ Error: Missing required argument
│ 
│   on .terraform/modules/vault/gke.tf line 69, in resource "google_container_node_pool" "vault":
│   69:     workload_metadata_config {
│ 
│ The argument "mode" is required, but no definition was found.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/vault/gke.tf line 70, in resource "google_container_node_pool" "vault":
│   70:       node_metadata = var.workload_identity_enable ? "GKE_METADATA_SERVER" : "SECURE"
│ 
│ An argument named "node_metadata" is not expected here.
╵
╷
│ Error: Unsupported attribute
│ 
│   on .terraform/modules/vault/kms.tf line 17, in resource "google_kms_crypto_key" "unseal":
│   17:   key_ring = google_kms_key_ring.vault.self_link
│ 
│ This object has no argument, nested block, or exported attribute named
│ "self_link".
╵
╷
│ Error: Unsupported attribute
│ 
│   on .terraform/modules/vault/kms.tf line 30, in resource "google_kms_crypto_key" "storage":
│   30:   key_ring = google_kms_key_ring.vault.self_link
│ 
│ This object has no argument, nested block, or exported attribute named
│ "self_link".
╵
╷
│ Error: Unsupported attribute
│ 
│   on .terraform/modules/vault/outputs.tf line 8, in output "key_ring_self_link":
│    8:   value       = google_kms_key_ring.vault.self_link
│ 
│ This object has no argument, nested block, or exported attribute named
│ "self_link".
```

**Comment by lawliet89** [comment]
https://github.com/basisai/terraform-google-vault/pull/11 constrains the provider version to `< 4` and will be released as `v0.7.1`.

**Comment by lawliet89** [comment]
https://github.com/basisai/terraform-google-vault/pull/12 will contain breaking changes to support `>= 4.0` of the Google provider.
